amundsen_application/static/js/ducks/search/reducer.ts

```diff
@@ -58,19 +58,26 @@ const initialState: SearchReducerState = {
 };
 
 export default function reducer(state: SearchReducerState = initialState, action: SearchReducerAction): SearchReducerState {
-  const newState = action.payload;
   switch (action.type) {
+    // Updates search term to reflect action
```

Comment:
I notice a side effect of this is that since the search team gets updated when the request is made, the component will re-render because components connected to Redux will re-render when their properties are updated. As a result the component will temporarily render in a state where it has as search term, but no search results. This leads to the user seeing a brief flash of: "Your search -- <searchTerm> -- did not match any tables result".

There are many ways we can address this, from rethinking our logic to introducing a loading state on the `SearchPage` in the same way that it exists on the `TableDetailPage`. I'm in favor of introducing the loading state in order to make these pages consistent, which we can do as a follow up PR.

Reply:
Ah, yeah was going to mention this, I think a loading state would make the most amount of sense here. Should I just file a follow up task?

Reply:
Already done, you should see them assigned to you in our internal queue.

Reply:
:+1: